pkg/virtctl/vm/expand_test.go

```diff
@@ -21,200 +21,177 @@ package vm_test
+		cmd := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_EXPAND, "--file", "invalid/path")
+		Expect(cmd()).To(MatchError("error reading file open invalid/path: no such file or directory"))
 	})
 
-	Context("with file input", func() {
-		var file *os.File
-
-		BeforeEach(func() {
-			var err error
-			file, err = os.CreateTemp(GinkgoT().TempDir(), "file-*")
-			Expect(err).ToNot(HaveOccurred())
-		})
-
-		It("should succeed when called with valid file input", func() {
-			Expect(os.WriteFile(file.Name(), []byte(vmSpec), 0666)).To(Succeed())
-			Expect(yaml.Unmarshal([]byte(vmSpec), vm)).To(Succeed())
-
-			kubecli.MockKubevirtClientInstance.EXPECT().ExpandSpec(k8smetav1.NamespaceDefault).Return(expandSpecInterface).Times(1)
-			expandSpecInterface.EXPECT().ForVirtualMachine(vm).Return(vm, nil).Times(1)
+	DescribeTable("should succeed when called on vm with", func(extraArgs []string) {
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return((virtClient.KubevirtV1().VirtualMachines(k8smetav1.NamespaceDefault))).Times(1)
+		_, err := virtClient.KubevirtV1().VirtualMachines(k8smetav1.NamespaceDefault).Create(context.Background(), kubecli.NewMinimalVM(vmName), k8smetav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
-			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, file.Name())
-			err := cmd()
+		args := append([]string{virtctl.COMMAND_EXPAND, "--vm", vmName}, extraArgs...)
+		cmd := clientcmd.NewRepeatableVirtctlCommand(args...)
 
-			Expect(err).ToNot(HaveOccurred())
-		})
+		Expect(cmd()).To(Succeed())
+		Expect(kvtesting.FilterActions(&virtClient.Fake, "get", "virtualmachines", "expand-spec")).To(HaveLen(1))
+	},
+		Entry("implicit default format", nil),
+		Entry("supported format json", []string{"--output", "json"}),
+		Entry("supported format yaml", []string{"--output", "yaml"}),
+	)
 
-		It("should fail when called with invalid vm defined in file input", func() {
-			Expect(os.WriteFile(file.Name(), []byte(vmSpec), 0666)).To(Succeed())
-			Expect(yaml.Unmarshal([]byte(vmSpec), vm)).To(Succeed())
+	Context("expanding a vm spec", func() {
+		const invalidYaml = `apiVersion: kubevirt.io/v1kind: VirtualMachine`
 
-			kubecli.MockKubevirtClientInstance.EXPECT().ExpandSpec(k8smetav1.NamespaceDefault).Return(expandSpecInterface).Times(1)
-			expandSpecInterface.EXPECT().ForVirtualMachine(vm).Return(nil, fmt.Errorf("error expanding vm from file")).Times(1)
+		var expandSpecInterface *kubecli.MockExpandSpecInterface
+		var vm *v1.VirtualMachine
+		var vmSpec []byte
 
-			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, file.Name())
-			err := cmd()
+		BeforeEach(func() {
+			expandSpecInterface = kubecli.NewMockExpandSpecInterface(ctrl)
+			vm = kubecli.NewMinimalVM(vmName)
 
-			Expect(err).To(HaveOccurred())
-			Expect(err).Should(MatchError("error expanding VirtualMachine - testvm in namespace - default: error expanding vm from file"))
+			var err error
```

Comment:
nit: move to the top. Like:

```go
BeforeEach(func() {
  var err error
  expandSpecInterface = kubecli.NewMockExpandSpecInterface(ctrl)
  vm = kubecli.NewMinimalVM(vmName)
[...]
```

Reply:
I wanted to keep it in the place where it is needed. (which is the next line)